### ISSUE TITLE
loading a .d.tl file with wrong casing loads file as other type

### ISSUE THREAD

**Issue by lenscas** [teal-language/tl#239]
when loading a .d.tl file while using the wrong casing it loads the file without giving an error. However, the types inside of it are seen as different types compared to when it is loaded with the correct casing. This combined with #198 gives rather confusing and hard to spot errors.

For example:

type.d.tl:
```lua
local record Test2
    a : number
end
return Test2
```
test1.tl
```lua
local Test2 = require("type")
return function(a:Test2)
    print(a)
end
```
test2.tl
```lua
local Test2 = require("Type")  --note how this starts with a capital T
local tes1 = require("tes1")
local a : Test2 = {a=1}
tes1(a)
```

This gives the error `argument 1: Test2 is not a Test2`

**Comment by pdesaulniers** [comment]
> when loading a .d.tl file while using the wrong casing it loads the file without giving an error

I cannot reproduce:

```
tl check test2.tl 
========================================
5 errors:
test2.tl:1:15: module not found: 'Type'
test2.tl:2:14: module not found: 'tes1'
test2.tl:3:11: unknown type Test2
test2.tl:3:19: in local declaration: a: got {a: number}, expected Test2
test2.tl:4:1: not a function: <unknown type>
```

**Comment by lenscas** [comment]
Then I guess its a problem with the version on luarocks and has been fixed since that release?

**Comment by lenscas** [comment]
just tried master (or more precise the branch for https://github.com/teal-language/tl/pull/243 but I doubt that that matters.)

```
test_teal_bug ../tl/tl check test2.tl 
========================================
1 error:
test2.tl:4:6: argument 1: Test2 is not a Test2
```

**Comment by hishamhm** [comment]
> I'm currently running it on MacOs, I can probably give it a try on Linux at a later date.

This might have something to do with it, since some filesystems on the Mac are case-insensitive (but case-preserving), and Teal keeps a mapping between module names and filenames IIRC.

**Comment by hishamhm** [comment]
Revisiting this issue, given what I just implemented in #264, I think the error message will become `Type2 (defined in type.d.tl:1) is not a Type2 (defined in Type.d.tl:1)`, which at least points to the filesystem case-insensitivity being the culprit. I don't think this is something we should try to work around, and instead the programmer needs to be consistent with the casing when using `require`, because at runtime Lua will have the same issue (loading the module twice, into two different entries of `package.loaded`).

So I will be closing this one, and at least we understand precisely what's going on!
